src/pykeen/training/lcwa.py

```diff
@@ -237,3 +225,65 @@ def _check_slicing_availability(self, supports_sub_batching: bool):
             report = "This model doesn't support sub-batching and slicing is not" " implemented for this model yet."
         logger.warning(report)
         raise MemoryError("The current model can't be trained on this hardware with these parameters.")
+
+
+# note: we use Tuple[Tensor] here, so we can re-use TensorDataset instead of having to create a custom one
+class SymmetricLCWATrainingLoop(TrainingLoop[Tuple[MappedTriples], Tuple[MappedTriples]]):
+    """
+    A "symmetric" LCWA scoring heads *and* tails at once.
+
+    .. seealso:: [lacroix2018]
+    """
+
+    # docstr-coverage: inherited
+    def _create_training_data_loader(
+        self, triples_factory: CoreTriplesFactory, sampler: Optional[str], **kwargs
+    ) -> DataLoader[Tuple[MappedTriples]]:  # noqa: D102
+        assert sampler is None
+        return DataLoader(dataset=TensorDataset(triples_factory.mapped_triples), **kwargs)
+
+    # docstr-coverage: inherited
+    def _process_batch(
+        self,
+        batch: Tuple[MappedTriples],
+        start: int,
+        stop: int,
+        label_smoothing: float = 0,
+        slice_size: Optional[int] = None,
+    ) -> torch.FloatTensor:  # noqa: D102
+        # unpack
+        hrt_batch = batch[0]
```

Comment:
Im confused by this line, why is only one thing unpacked?

Reply:
https://github.com/pykeen/pykeen/pull/1098#discussion_r958197972